### ISSUE TITLE
Rule idea: calculations-braces-around

### ISSUE THREAD

**Issue by dryoma** [kristerkari/stylelint-scss#21]
Inspired by [Sass Guilelines](https://sass-guidelin.es/#calculations)

Top-level numeric calculations should always be wrapped in parentheses. Not only does this requirement dramatically improve readability, it also prevents some edge cases by forcing Sass to evaluate the contents of the parentheses.

```css
// Ok
.foo {
  width: (100% / 3);
}

// not ok
.foo {
  width: 100% / 3;
}
```

**Comment by kristerkari** [comment]
Sounds good, the only thing that I'm worried about is that this kind of rule could have a lot of edge cases when linting. 

I did a bit similar stuff to Stylefmt to format code with parens, so I know that it is not easy to get right:
e.g. https://github.com/morishitter/stylefmt/pull/70

**Comment by dryoma** [comment]
Well, yeah...

I just decided to drop all the more-or-less decsent ideas here instead of piling then in a local .txt document :)

**Comment by jeddy3** [comment]
`calculation-no-missing-surrounding-parentheses`?

**Comment by evilebottnawi** [comment]
@jeddy3 here options: "always" and "never" be good here too. Some like add parentheses, some not like.

**Comment by jeddy3** [comment]
Sure. I wasn't really following [my own advice](https://github.com/kristerkari/stylelint-scss/issues/39#issuecomment-226234641) when I came up with that name suggestion... whoops.

**Comment by dryoma** [comment]
I hate to admit it, but more and more I doubt about this rule. Particularly, after working on `operators-no-unspaced` and figuring out all edge cases of what SCSS considers an operation.

For exapmle,
```scss
a {
  width: (10px +1); // compiles into an operation: `width: 11px;`
  width: 10px +1; // compiles into a list `width: 10px +1`
}
```
Right now both Sass compilers and our [parsing helpers](https://github.com/kristerkari/stylelint-scss/blob/operator-spaces/src/utils/sassValueParser/index.js) consider the secont as a list indeed. But how do we know if it is *intended* to be a list? Of cource in the example above the secont decl is obviousely malformed, yet... To be honest, I don't fancy the idea of analysing the semantics of the SCSS code to *quess* if what is not an operation at the moment is really intended to be one.

Any thoughts?

**Comment by jeddy3** [comment]
> I hate to admit it, but more and more I doubt this rule will be possible to implement. 

I understand what you mean.

> Any thoughts?

I feel the same way [as David](http://davidtheclark.com/lessons-i-may-have-learned/) about how it might not be worth the effort in trying to work around the complexities of this type of syntax. My suggestion would be focus on the rules that check SCSS-like language extensions that more closely conform to standard CSS syntax. That way this plugin could be part of a shift towards using SCSS-like processors based on only standard CSS syntax. The alternatively is to put a lot of energy into solving a problem that can be more easily solved another way i.e. only using PostCSS plugins that conform to standard CSS syntax e.g. in this case wrapping those math operations in a `calc()` function that is reduced down at compile time.

**Comment by kristerkari** [comment]
> I hate to admit it, but more and more I doubt this rule will be possible to implement. Particularly, after working on operators-no-unspaced and figuring out all those edge cases of what SCSS considers an operation.

That's what I was afraid of. Should we drop this rule @dryoma ?

**Comment by dryoma** [comment]
Let's remove it from the milestones list for now. I'll try to give it a thorough thought later again (or maybe someone else comes up with some good suggestion).
